catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py

```diff
@@ -107,6 +107,63 @@ def test_get_query_param_offset_page_number():
     assert actual_param == expected_param
 
 
+@pytest.mark.parametrize(
+    "record",
+    [
+        # Missing any item in the foreign_landing_url chain
+        # Here we return `None` only if the foreign_landing_url is missing.
+        # Otherwise, we create an image list, check each image's
+        # required fields and return a blank list if none are valid (see next test).
+        {"links": {}},
+        {"links": {"self": ""}},
+        {"links": {"self": "link"}, "attributes": {"multimedia": []}},
+    ],
+)
+def test_get_record_data_returns_none_for_falsy_foreign_landing_url_and_multimedia(
+    record,
+):
+    actual_record_data = sm.get_record_data(record)
+    assert actual_record_data is None
```

Comment:
It's more difficult to parametrize these tests because of the different nesting levels: we would have to somehow get the correct nested property and set it to null. I've used the order in which we extract the properties here, so that `foreign_landing_url` is the first thing that is checked in the provider script, and the test will return `None` right away. For other tests, `foreign_landing_url` is always available because they first check for it before checking other parameters.

We could create a separate issue to improve the tests here, what do you think?